.github/scripts/check-patterns.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+# Enable globstar for recursive globbing
+shopt -s globstar
+
+ALL_CHANGED_FILES="$1"
+IFS=' ' read -r -a changed <<< $ALL_CHANGED_FILES
+
+TOOL_CONFIG_PATTERNS=("docker/tool-config.Dockerfile .github/workflows/tool-config.yml docker/conf/**")
+BUILDER_2004_PATTERNS=("docker/ubuntu-2004-builder.Dockerfile .github/workflows/builder-ubuntu.yml")
+BUILDER_2204_PATTERNS=("docker/ubuntu-2204-builder.Dockerfile .github/workflows/builder-ubuntu.yml")
+CONTRIB_PATTERNS=("docker/psibase-contributor.Dockerfile .github/workflows/contributor.yml")
+
+matches_pattern() {
+    local pattern="$1"
+    for file in ${changed[@]}; do
+        if [[ $file == $pattern ]]; then
+            return 0 # Success
+        fi
+    done
+    return 1
+}
+
+# Possible dispatching cases and their return values
+# 1: run tool config (and all dependent workflows)
+# 2: run 2004 builder
```

Comment:
I don't have CI/CD build that, haven't for a while. For now it's just the builder since we still try to confirm in psibase ci/cd that it builds on both.

Reply:
I meant if both ubuntu-2004 and psibase-contributor have (independent) modifications. It looks to me like the update to psibase-contributor gets dropped on the floor in this case.

Reply:
Fixed